openedx/core/djangoapps/course_groups/views.py

```diff
@@ -99,15 +125,40 @@ def course_cohort_settings_handler(request, course_key_string):
     This will raise 404 if user is not staff.
     GET
         Returns the JSON representation of cohort settings for the course.
-    PUT or POST
+    PUT, POST or PATCH
         Updates the cohort settings for the course. Returns the JSON representation of updated settings.
     """
     course_key = CourseKey.from_string(course_key_string)
-    get_course_with_access(request.user, 'staff', course_key)
-    if request.method == 'GET':
-        cohort_settings = cohorts.get_course_cohort_settings(course_key)
-        return JsonResponse(_get_course_cohort_settings_representation(cohort_settings))
-    else:
+    course = get_course_with_access(request.user, 'staff', course_key)
+    cohort_settings = cohorts.get_course_cohort_settings(course_key)
+
+    if request.method == 'PATCH':
+        cohorted_course_wide_ids, cohorted_inline_ids = get_cohorted_discussion_ids(
+            course,
+            cohort_settings.cohorted_discussions
+        )
+
+        if request.json.get('cohorted_course_wide_discussions'):
+            cohorted_inline_ids.extend(request.json.get('cohorted_course_wide_discussions'))
+            cohorted_discussions = cohorted_inline_ids
+        elif request.json.get('cohorted_inline_discussions'):
+            cohorted_course_wide_ids.extend(request.json.get('cohorted_inline_discussions'))
+            cohorted_discussions = cohorted_course_wide_ids
+        else:
+            cohorted_discussions = cohort_settings.cohorted_discussions
+
+        try:
+            cohort_settings = cohorts.set_course_cohort_settings(
+                course_key,
+                cohorted_discussions=cohorted_discussions,
+                always_cohort_inline_discussions=request.json.get(
+                    'always_cohort_inline_discussions', cohort_settings.always_cohort_inline_discussions
+                )
+            )
+        except ValueError as err:
+            # Note: error message not translated because it is not exposed to the user (UI prevents this state).
+            return JsonResponse({"error": unicode(err)}, 400)
+    elif request.method in ["PUT", "POST"]:
         is_cohorted = request.json.get('is_cohorted')
         if is_cohorted is None:
```

Comment:
Using `PUT` or `POST` just for setting `is_cohorted` is no longer consistent with the semantics of these methods. This should be changed to use `PATCH` too.